dotCMS/src/main/java/com/dotmarketing/cms/urlmap/URLMapAPIImpl.java

```diff
@@ -268,34 +256,40 @@ private void checkContentPermission(final UrlMapContext context, final Contentle
                 contentlet, PermissionLevel.READ.getType(), context.getUser(), context.getMode().respectAnonPerms);
 
         if (!havePermission) {
-            throw new DotSecurityException(String.format("User dont have permission in content: %s", contentlet.getName()));
+            throw new DotSecurityException(String.format("User does not have permission in content: %s", contentlet.getName()));
         }
     }
 
     private String buildContentQuery(
             final Matches matches,
-            final Structure structure,
-            final Field hostField,
+            final ContentType structure,
             final UrlMapContext context) {
 
         final StringBuilder query = new StringBuilder();
 
-        query.append("+structureName:").append(structure.getVelocityVarName())
-             .append(" +deleted:false ");
-
-        if (context.getMode() == PageMode.PREVIEW_MODE || context.getMode() == PageMode.EDIT_MODE) {
-            query.append("+working:true ");
+        query.append("+contentType:")
+            .append(structure.variable())
+            .append(" +deleted:false ")
+            .append(" +(conhost:")
+                .append(context.getHost().getIdentifier())
+                .append(" OR conhost:")
+                .append(Host.SYSTEM_HOST)
+            .append(")");
```

Comment:
Codacy found an issue: [Avoid appending characters as strings in StringBuffer.append.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5796110)